src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -25,38 +25,14 @@ static inline int MPIDI_OFI_win_allgather(MPIR_Win * win, void *base, int disp_u
     uint32_t first;
     MPIR_Errflag_t errflag = MPIR_ERR_NONE;
     MPIR_Comm *comm_ptr = win->comm_ptr;
-    int raw_prefix, idx, bitpos;
-    unsigned gen_id;
     MPIDI_OFI_win_targetinfo_t *winfo;
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_WIN_ALLGATHER);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_WIN_ALLGATHER);
 
-    /* Calculate a canonical context id */
-    raw_prefix = MPIR_CONTEXT_READ_FIELD(PREFIX, comm_ptr->context_id);
-    idx = raw_prefix / MPIR_CONTEXT_INT_BITS;
-    bitpos = raw_prefix % MPIR_CONTEXT_INT_BITS;
-    gen_id = (idx * MPIR_CONTEXT_INT_BITS) + (31 - bitpos);
-
-    int total_bits_avail = MPIDI_Global.max_mr_key_size * 8;
-    uint64_t window_instance = (uint64_t) (MPIDI_OFI_WIN(win).win_id) >> 32;
-    int bits_for_instance_id = MPIDI_Global.max_windows_bits;
-    int bits_for_context_id;
-    uint64_t max_contexts_allowed;
-    uint64_t max_instances_allowed;
-
-    bits_for_context_id = total_bits_avail -
-        MPIDI_Global.max_windows_bits - MPIDI_Global.max_huge_rma_bits;
-    max_contexts_allowed = (uint64_t) 1 << (bits_for_context_id);
-    max_instances_allowed = (uint64_t) 1 << (bits_for_instance_id);
-    MPIR_ERR_CHKANDSTMT(gen_id >= max_contexts_allowed, mpi_errno, MPI_ERR_OTHER,
-                        goto fn_fail, "**ofid_mr_reg");
-    MPIR_ERR_CHKANDSTMT(window_instance >= max_instances_allowed, mpi_errno, MPI_ERR_OTHER,
-                        goto fn_fail, "**ofid_mr_reg");
```

Comment:
In the new code, equivalent to these error checks are gone. I think it would be nice to add them back in `win_init` or somewhere.

Reply:
It is moved into `MPIDI_OFI_win_init`